packages/editor/src/components/table-of-contents/panel.js

```diff
@@ -5,13 +5,22 @@ import { Fragment } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 import { withSelect } from '@wordpress/data';
 
+import { delay } from 'lodash';
+
 /**
  * Internal dependencies
  */
 import WordCount from '../word-count';
 import DocumentOutline from '../document-outline';
 
-function TableOfContentsPanel( { headingCount, paragraphCount, numberOfBlocks } ) {
+function TableOfContentsPanel( { headingCount, paragraphCount, numberOfBlocks, onRequestClose } ) {
+	// Remove the block id hash from the url because block ids are only valid for the session.
+	const removeBlockIdHash = () => {
+		delay( () => {
+			const { pathname, search } = window.location;
+			window.history.pushState( '', document.title, pathname + search );
```

Comment:
`pushState` would add to the browser history, which I don't think is desirable for this interaction. `replaceState` seems more appropriate. Would you agree?

Reply:
ah right, good point.

Reply:
addressed in https://github.com/WordPress/gutenberg/pull/10815/commits/bada58fe6e0b180e36172f4800c8765bfec76766